instrumentation/nocode/src/main/java/com/splunk/opentelemetry/instrumentation/nocode/YamlParser.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.instrumentation.nocode;
+
+import com.splunk.opentelemetry.javaagent.bootstrap.nocode.NocodeRules;
+import java.io.FileReader;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import org.snakeyaml.engine.v2.api.Load;
+import org.snakeyaml.engine.v2.api.LoadSettings;
+
+public class YamlParser {
+  private static final Logger logger = Logger.getLogger(YamlParser.class.getName());
+  // FIXME support method override selection - e.g., with classfile method signature or something
+  public static final String NOCODE_YMLFILE_ENV_KEY = "SPLUNK_OTEL_INSTRUMENTATION_NOCODE_YML_FILE";
+
+  private final List<NocodeRules.Rule> InstrumentationRules;
+
+  public YamlParser() {
+    InstrumentationRules = Collections.unmodifiableList(new ArrayList<>(load()));
+  }
+
+  public List<NocodeRules.Rule> getInstrumentationRules() {
+    return InstrumentationRules;
+  }
+
+  private static List<NocodeRules.Rule> load() {
+    try {
+      return loadUnsafe();
+    } catch (Exception e) {
+      logger.severe("Can't load configured yaml: " + e);
```

Comment:
Prefer to keep the exception details rather than `toString()`. Also added the word `nocode` to the message for context.

```suggestion
      logger.log(Level.SEVERE, "Can't load configured nocode yaml.", e)
```